lib/item_bloc.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:bloc/bloc.dart';
+import 'package:bloc_sample/main.dart';
+
+/// Manages Actions/Business Logic for a single item
+class ItemBloc extends Bloc<ItemEvent, ItemState> {
+  @override
+  ItemState get initialState => ItemInitial();
+
+  @override
+  Stream<ItemState> mapEventToState(ItemEvent event) async* {
+    if (event is AssignItem) {
+      if (event.type == ItemType.Private) {
+        try {
+          // assign item (call repo)
+          // yield list with updated item
```

Comment:
Assigning an item will change the item data - therefore the list item needs to be updated.
So I'd need to again emit item data with the Success event?